CharNGram link broken

## Test Failure

```
self = <torchtext_unittest.test_build.TestVocab testMethod=test_download_charngram_vectors>

    def test_download_charngram_vectors(self) -> None:
        # Build a vocab and get vectors twice to test caching.
        for _ in range(2):
>           vectors = torchtext.vocab.CharNGram()

torchtext_unittest/test_build.py:70: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../torchtext/vocab/vectors.py:239: in __init__
    super(CharNGram, self).__init__(self.name, url=self.url, **kwargs)
../torchtext/vocab/vectors.py:59: in __init__
    self.cache(name, cache, url=url, max_vectors=max_vectors)
../torchtext/vocab/vectors.py:95: in cache
    urlretrieve(url, dest, reporthook=reporthook(t))
../ci_env/lib/python3.8/urllib/request.py:247: in urlretrieve
    with contextlib.closing(urlopen(url, data)) as fp:
../ci_env/lib/python3.8/urllib/request.py:222: in urlopen
    return opener.open(url, data, timeout)
../ci_env/lib/python3.8/urllib/request.py:531: in open
    response = meth(req, response)
../ci_env/lib/python3.8/urllib/request.py:640: in http_response
    response = self.parent.error(
../ci_env/lib/python3.8/urllib/request.py:563: in error
    result = self._call_chain(*args)
../ci_env/lib/python3.8/urllib/request.py:502: in _call_chain
    result = func(*args)
../ci_env/lib/python3.8/urllib/request.py:755: in http_error_302
    return self.parent.open(new, timeout=req.timeout)
../ci_env/lib/python3.8/urllib/request.py:531: in open
    response = meth(req, response)
../ci_env/lib/python3.8/urllib/request.py:640: in http_response
    response = self.parent.error(
../ci_env/lib/python3.8/urllib/request.py:569: in error
    return self._call_chain(*args)
../ci_env/lib/python3.8/urllib/request.py:502: in _call_chain
    result = func(*args)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <urllib.request.HTTPDefaultErrorHandler object at 0x7f805dec1b80>
req = <urllib.request.Request object at 0x7f805decc2e0>
fp = <http.client.HTTPResponse object at 0x7f805decc0d0>, code = 404
msg = 'Not Found', hdrs = <http.client.HTTPMessage object at 0x7f805decc580>

    def http_error_default(self, req, fp, code, msg, hdrs):
>       raise HTTPError(req.full_url, code, msg, hdrs, fp)
E       urllib.error.HTTPError: HTTP Error 404: Not Found

../ci_env/lib/python3.8/urllib/request.py:649: HTTPError
```

## Screenshot of 404 page

<img width="1245" alt="Screenshot 2023-03-08 at 2 29 58 PM" src="https://user-images.githubusercontent.com/20175092/223824252-0a0a1e48-3c7b-499c-9e59-fa30537ae0cc.png">

--- a/test/torchtext_unittest/common/torchtext_test_case.py
+++ b/test/torchtext_unittest/common/torchtext_test_case.py
@@ -18,7 +18,7 @@ def third_party_download(test_func):
         try:
             return test_func(*args, **kwargs)
         except HTTPError as e:
-            logger.warning(e)
+            logger.warning(f"Cannot access URL in {test_func.__name__}. Error message {e}")
 
     return inner
 

--- a/test/torchtext_unittest/common/torchtext_test_case.py
+++ b/test/torchtext_unittest/common/torchtext_test_case.py
@@ -5,11 +5,22 @@ import os
 import shutil
 import subprocess
 import tempfile
+from urllib.error import HTTPError
 
 import torch  # noqa: F401
 from torch.testing._internal.common_utils import TestCase
 
 logger = logging.getLogger(__name__)
+
+
+def third_party_download(test_func):
+    def inner(*args, **kwargs):
+        try:
+            return test_func(*args, **kwargs)
+        except HTTPError as e:
+            logger.warning(e)
+
+    return inner
 
 
 class TorchtextTestCase(TestCase):

--- a/test/torchtext_unittest/test_build.py
+++ b/test/torchtext_unittest/test_build.py
@@ -5,7 +5,7 @@ import os
 import torch
 import torchtext.data
 
-from .common.torchtext_test_case import TorchtextTestCase
+from .common.torchtext_test_case import TorchtextTestCase, third_party_download
 
 
 class TestDataUtils(TorchtextTestCase):
@@ -64,6 +64,7 @@ class TestVocab(TorchtextTestCase):
         self.assertEqual(token_one_vec.shape[0], vec.dim)
         self.assertEqual(vec[tokens[0].lower()], token_one_vec)
 
+    @third_party_download
     def test_download_charngram_vectors(self) -> None:
         # Build a vocab and get vectors twice to test caching.
         for _ in range(2):

